In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [2]:
data = pd.read_csv("Health_Insurance.csv")
data.head()

Age  Purchased
0   19          0
1   35          0
2   26          0
3   27          0
4   19          0

# Seperate X and y

In [3]:
X = data.drop(columns = ['Purchased'])
y = data['Purchased']

# Apply Logistic Regression on the data

In [4]:
from sklearn.linear_model import LogisticRegression
lr = LogisticRegression()
lr

LogisticRegression()

# Applying Logistic Regression on the 5 Fold Cross validation data

In [5]:
# cross_val_score(ML algorithm, X & y i.e your data, K Fold Cross validation for multiple train and test 
# Split from the data)

In [6]:
from sklearn.model_selection import cross_val_score
cross_val_score(lr, X, y, cv = 5)

array([0.8   , 0.875 , 0.85  , 0.7625, 0.825 ])

- 1st train and test set = 80%
- 2nd train and test set = 87.5%
- 3rd train and test set = 85%
- 4th train and test set = 76.25%
- 5th train and test set = 82.5%

- Out of all the accuracies the 2nd train and test set is giving the best accuracy of 87.5%

In [7]:
from sklearn.model_selection import cross_val_score
cross_val_score(lr, X, y, cv = 10)

array([0.95 , 0.65 , 0.825, 0.925, 0.925, 0.825, 0.75 , 0.75 , 0.8  ,
       0.975])

# Which Fold  Cross Validation is the best?
- Generally, 10 Fold cross validation is prefered across the industry. 

# If you want to have a look at the 2nd train and test data which is giving the highest accuray when we performed 5 Fold Cross validation

In [9]:
from sklearn.model_selection import KFold
kf = KFold(n_splits = 5)
kf

KFold(n_splits=5, random_state=None, shuffle=False)

In [10]:
kf.split(X, y)

<generator object _BaseKFold.split at 0x000001EEA7631DD0>

In [11]:
list(kf.split(X, y))

[(array([ 80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,  91,  92,
          93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103, 104, 105,
         106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118,
         119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131,
         132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144,
         145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157,
         158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170,
         171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183,
         184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196,
         197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209,
         210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222,
         223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235,
         236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248,
         249, 250, 251, 252, 253, 254,